src/gui/properties/speedplotview.h

```diff
@@ -76,14 +76,32 @@ class SpeedPlotView: public QGraphicsView
     void setGraphEnable(GraphID id, bool enable);
     void setViewableLastPoints(TimePeriod period);
 
-    void pushPoint(PointData point);
+    void pushPoint(const PointData &point);
 
     void replot();
 
 protected:
     virtual void paintEvent(QPaintEvent *event);
 
 private:
+    typedef boost::circular_buffer<PointData> PointBufferType;
+    class Divider
```

Comment:
I don't like both confusing name of this class and its design. As well as surrounding code (including previous code).

Reply:
Tell me better name for the class and jobs is done :) . 
Of course, my code is not a tip top of code engineering skill, and I'm not fully satisfied with it too. There left too many constants, data buffer for 1min and 5 min period is shared and it adds a mess. It looks better for me to incorporate circular buffer into my Divider, rename it to DataBackend, and use uniformly to any period of observation, but it require to rewrite a lot of the code. but my PR  already expands uncontrollable :( .